Add more smart pointers to accessibility

abigailfox opened the pull request:
#### acd3844b0421eb9a9780d21e8a5a9255c0eae3ce
<pre>
Add more smart pointers to accessibility
<a href="https://bugs.webkit.org/show_bug.cgi?id=271411">https://bugs.webkit.org/show_bug.cgi?id=271411</a>
<a href="https://rdar.apple.com/125190532">rdar://125190532</a>

Reviewed by Ryosuke Niwa.

Memory safety improvements for  AccessibilityScrollbar.cpp, AccessibilitySlider.cpp, AccessibilityTableColumn.cpp, AccessibilityTableRow.cpp, and AccessibilityTree.cpp

* Source/WebCore/accessibility/AccessibilityScrollbar.cpp:
(WebCore::AccessibilityScrollbar::document const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::addChildren):
(WebCore::AccessibilitySlider::setValue):
* Source/WebCore/accessibility/AccessibilityTableColumn.cpp:
(WebCore::AccessibilityTableColumn::addChildren):
* Source/WebCore/accessibility/AccessibilityTableRow.cpp:
(WebCore::AccessibilityTableRow::rowHeader):
* Source/WebCore/accessibility/AccessibilityTree.cpp:
(WebCore::AccessibilityTree::isTreeValid const):

Canonical link: <a href="https://commits.webkit.org/276528@main">https://commits.webkit.org/276528@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ba8681ea6429455405b940b587aa1b2f452addb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44885 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23984 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47375 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47539 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40890 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/47188 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/28079 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21381 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45463 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21030 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38642 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17936 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18442 "Unexpected infrastructure issue, retrying build") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39786 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2932 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41110 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40079 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49210 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19853 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16394 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43845 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21172 "Built successfully") | | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42607 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/proxy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9993 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/21510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20846 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebCore/accessibility/AccessibilityTree.cpp:
```diff
@@ -82,23 +82,26 @@ bool AccessibilityTree::isTreeValid() const
         return false;
+
+    // This function does not require smart pointers as long as we ensure it does not modify the DOM
+    ASSERT(ScriptDisallowedScope::InMainThread::isEventDispatchAllowedInSubtree(node));
```
No need for this comment, since you've added the RefPtrs.

abigailfox replied:
Oops, good catch!

geoffreygaren commented on Source/WebCore/accessibility/AccessibilityScrollbar.cpp:
```diff
@@ -60,7 +60,7 @@ Document* AccessibilityScrollbar::document() const
     if (!parent)
         return nullptr;
-    return parent->document();
+    return parent->protectedDocument().get();
```
Can you share the tool output that motivated this change? Both lines of code have the same pointer lifetime, so something seems to be amiss here. Maybe it's a bug in the tool.

rniwa replied:
Yeah, this doesn't really buy us anything since we're turning the raw pointer at the end of the day.

abigailfox replied:
Would it be more worthwhile to convert `AccessibilityObject* parent = parentObject();` to `RefPtr parent = parentObject();`?

rniwa replied:
No. There is no point in deploying smart pointer here given we're not calling anything but a trivial function.

abigailfox committed:
Add more smart pointers to accessibility
https://bugs.webkit.org/show_bug.cgi?id=271411
rdar://125190532

Reviewed by Ryosuke Niwa.

Memory safety improvements for  AccessibilityScrollbar.cpp, AccessibilitySlider.cpp, AccessibilityTableColumn.cpp, AccessibilityTableRow.cpp, and AccessibilityTree.cpp

* Source/WebCore/accessibility/AccessibilityScrollbar.cpp:
(WebCore::AccessibilityScrollbar::document const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::addChildren):
(WebCore::AccessibilitySlider::setValue):
* Source/WebCore/accessibility/AccessibilityTableColumn.cpp:
(WebCore::AccessibilityTableColumn::addChildren):
* Source/WebCore/accessibility/AccessibilityTableRow.cpp:
(WebCore::AccessibilityTableRow::rowHeader):
* Source/WebCore/accessibility/AccessibilityTree.cpp:
(WebCore::AccessibilityTree::isTreeValid const):

Canonical link: https://commits.webkit.org/276528@main
